src/Reflection/ClassInspector.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bigcommerce\Injector\Reflection;
+
+use Bigcommerce\Injector\Cache\ServiceCacheInterface;
+use ReflectionClass;
+use ReflectionException;
+
+class ClassInspector
+{
+    public function __construct(
+        private readonly ReflectionClassCache $reflectionClassCache,
+        private readonly ParameterInspector $parameterInspector,
+        private readonly ServiceCacheInterface $serviceCache,
+        private readonly ClassInspectorStats $stats
+    ) {
+    }
+
+    /**
+     * @param string $class
+     * @param string $method
+     * @return void
+     * @throws ReflectionException
+     */
+    public function inspectMethod(string $class, string $method): void
+    {
+        if ($this->classHasMethod($class, $method)) {
+            if ($this->methodIsPublic($class, $method)) {
+                $this->getMethodSignature($class, $method);
+            }
+        }
+    }
+
+    /**
+     * @template T of object
+     * @param string $class
+     * @param class-string<T> $class
+     * @param string $method
+     * @return bool The reflection class instance for the given class.
+     * @throws ReflectionException
+     */
+    public function classHasMethod(string $class, string $method): bool
+    {
+        $key = "$class::{$method}_exists";
```

Comment:
I would recommend using something like `$class::{$method}::exists`, just to avoid confusion with method names.

Reply:
That makes senes. Done.